juju_verify/verifiers/neutron_gateway.py

```diff
@@ -0,0 +1,162 @@
+# Copyright 2021 Canonical Limited.
+#
+# This file is part of juju-verify.
+#
+# juju-verify is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# juju-verify is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program. If not, see https://www.gnu.org/licenses/.
+"""neutron-gateway verification."""
+import json
+from typing import List
+
+from juju.unit import Unit
+
+from juju_verify.verifiers.base import BaseVerifier, Result, Severity
+from juju_verify.verifiers.result import aggregate_results
+from juju_verify.utils.action import data_from_action
+from juju_verify.utils.unit import parse_charm_name, run_action_on_unit
+
+
+class NeutronGateway(BaseVerifier):
+    """Implementation of verification checks for the neutron-gateway charm."""
+
+    NAME = 'neutron-gateway'
+    action_name_result_map = {"get-status-routers": "router-list",
+                              "get-status-dhcp": "dhcp-networks",
+                              "get-status-lb": "load-balancers"}
+    action_name_failure_string_map = {"get-status-routers": ("The following routers are "
+                                                             "non-redundant: {}"),
+                                      "get-status-dhcp": ("The following DHCP networks "
+                                                          "are non-redundant: {}"),
+                                      "get-status-lb": ("The following LBaasV2 LBs were "
+                                                        "found: {}. LBaasV2 does not "
+                                                        "offer HA.")}
+
+    @staticmethod
+    def get_unit_hostname(unit: Unit) -> str:
+        """Return name of the host on which the unit is running."""
+        get_hostname_action = run_action_on_unit(unit, "node-name")
+        hostname = data_from_action(get_hostname_action, "node-name")
+        return hostname
+
+    @classmethod
```

Comment:
Is it necessary to us a classmethod?

Reply:
It uses `cls` reference. Alternative is to have it `static` and use `NeutronGateway.action_name_result_map` which is worse imo.

Reply:
I was thinking to using classic class function with `self`. It's not used outside this class :eyes:.

Reply:
It does not use any instance attributes so the `self` is not really necessary. Anyway, the original UTs were written when this method was unbound function and it was easier to integrate this function into the class as a class method rather than classic method and rewrite the unit tests :D

Reply:
Ah, sorry I missed [this](https://github.com/mkalcok/juju-verify/blob/30cac1951c2e7526bba62ceab2889afda33f8cf0/tests/unit/verifiers/test_neutron_gateway.py#L328). I thought it was never called as a classmethod.